test/test-generate-pras.jl

```diff
@@ -0,0 +1,241 @@
+import DataFrames
+import Dates
+import Test
+import TimeSeries
+using Dates: DateTime
+
+const PSY = PowerSystems
+const PSCB = PowerSystemCaseBuilder
+
+"""
+Use PSCB to build RTS-GMLC System and add outage data
+as a Supplmental Attribute
+"""
+function get_rts_gmlc_outage()
+    rts_da_sys = PSCB.build_system(PSCB.PSISystems, "RTS_GMLC_DA_sys")
+
+    ###########################################
+    # Parse the gen.csv and add OutageData
+    # SupplementalAttribute to components for
+    # which we have this data
+    ###########################################
+    gen_for_data = CSV.read(joinpath(@__DIR__, "descriptors/gen.csv"), DataFrames.DataFrame)
+
+    for row in DataFrames.eachrow(gen_for_data)
+        λ, μ = PRASInterface.rate_to_probability(row.FOR, row["MTTR Hr"])
+        transition_data = PSY.GeometricDistributionForcedOutage(;
+            mean_time_to_recovery=row["MTTR Hr"],
+            outage_transition_probability=λ,
+        )
+        comp = PSY.get_component(PSY.Generator, rts_da_sys, row["GEN UID"])
+
+        if ~(isnothing(comp))
+            PSY.add_supplemental_attribute!(rts_da_sys, comp, transition_data)
+            @info "Added outage data supplemental attribute to $(row["GEN UID"]) generator"
+        else
+            @warn "$(row["GEN UID"]) generator doesn't exist in the System."
+        end
+    end
+    return rts_da_sys
+end
+
+function test_names_equal(
+    x::Vector{String},
+    y::Vector{String};
+    left="Left-side",
+    right="Right-side",
+)
+    x = sort(x)
+    y = sort(y)
+    idx = 1
+    while idx <= length(x) && idx <= length(y)
+        if x[idx] == y[idx]
+            idx += 1
+        elseif x[idx] != y[idx]
+            prev_idx = max(idx - 2, 1)
+            next_idx = min(idx + 2, length(x))
+            @error "$left $(x[prev_idx:next_idx]) != $right $(y[prev_idx:next_idx])"
+            return false
+        end
+    end
+    if idx > length(x) && idx > length(y)
+        return true
+    end
+    prev_idx = max(idx - 2, 1)
+    next_idx = min(idx + 2, length(x))
+    comparison = idx <= length(x) ? "more" : "less"
+    @error "$left has $comparison elements $(x[prev_idx:next_idx]) than $right $(y[prev_idx:next_idx])"
+    return false
+end
+
+function array_all_equal(x::AbstractVector{T}, v::T) where {T}
+    for (i, xi) in enumerate(x)
+        if !isapprox(xi, v)
+            @error "array[$i]: $xi != $v"
+            return false
+        end
+    end
+    return true
+end
+
+@testset "RTS GMLC" begin
+    rts_da_sys = get_rts_gmlc_outage()
+    area_names = PSY.get_name.(PSY.get_components(PSY.Area, rts_da_sys))
+    generator_names =
+        PSY.get_name.(
+            PSY.get_components(
+                x -> PSY.get_available(x) && PSY.get_rating(x) > 0,
+                Union{PSY.HydroDispatch, PSY.RenewableGen, PSY.ThermalGen},
+                rts_da_sys,
+            )
+        )
+    storage_names = PSY.get_name.(PSY.get_components(PSY.Storage, rts_da_sys))
```

Comment:
We need to figure out if we need to do the same thing I do in Sienna2PRAS to avoid double-counting of hybrid system sub components.

Reply:
Tracked in https://github.com/NREL-Sienna/PRASInterface.jl/issues/14